docs/content/guides/dagster/pythonic-config.mdx

```diff
@@ -0,0 +1,246 @@
+from dagster import asset
+from dagster._config.structured_config import Config
+
+class MyAssetConfig(Config):
+    person_name: str
+
+@asset
+def greeting(config: MyAssetConfig) -> str:
+    return f"hello {config.person_name}"
+```
+
+</TabItem>
+<TabItem name="Using ops and jobs">
+
+#### Using ops
+
+Here, we define a subclass of <PyObject object="Config"/> holding a single string value representing the name of a user. We can access the config through the `config` parameter in the op body.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_op_config endbefore=end_basic_op_config dedent=4
+from dagster import op
+from dagster._config.structured_config import Config
+
+class MyOpConfig(Config):
+    person_name: str
+
+@op
+def print_greeting(config: MyOpConfig):
+    print(f"hello {config.person_name}")
+```
+
+</TabItem>
+</TabGroup>
+
+### Defining and accessing Pythonic configuration for a resource
+
+Configurable parameters for a resource are defined by specifying attributes for a resource class, which subclasses <PyObject object="ConfigurableResource"/>. The below resource defines a configurable connection URL, which can be accessed in any methods defined on the resource.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_resource_config endbefore=end_basic_resource_config dedent=4
+from dagster import op
+from dagster._config.structured_config import ConfigurableResource
+
+class MyDatabaseResource(ConfigurableResource):
+    connection_url: str
+
+    def query(self, query: str):
+        return get_engine(self.connection_url).execute(query)
+```
+
+For more information on using resources, see [the Pythonic resources guide](/guides/dagster/pythonic-resources).
+
+### Specifying runtime configuration in code
+
+To execute a job or materialize an asset which specifies config, you'll need to provide values for its parameters. When specifying config from the Python API, we can use the `run_config` argument for <PyObject object="JobDefinition" method="execute_in_process"/> or <PyObject object="materialize"/>. This takes a `RunConfig` object, within which we can supply config on a per-op or per-asset basis. The config is specified as a dictionary, with the keys corresponding to the op/asset names and the values corresponding to the config values.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_execute_with_config endbefore=end_execute_with_config dedent=4
+from dagster import job, materialize, op
+
+from dagster._core.definitions.run_config import RunConfig
+
+@job
+def greeting_job():
+    print_greeting()
+
+job_result = greeting_job.execute_in_process(
+    run_config=RunConfig(ops={"print_greeting": MyOpConfig(person_name="Alice")})
+)
+
+asset_result = materialize(
+    [greeting],
+    run_config=RunConfig(ops={"greeting": MyAssetConfig(person_name="Alice")}),
```

Comment:
I see `assets=` in the preview. 

I would recommend omitting the `ops` keyword entirely here as it just adds confusion.

`run_config=RunConfig({"greeting": MyAssetConfig(person_name="Alice")})`